twake/backend/node/src/services/channels/web/controllers/channel.ts

```diff
@@ -486,6 +487,66 @@ export class ChannelCrudController
       }),
     );
   }
+
+  async recent(
+    request: FastifyRequest<{
+      Querystring: ChannelListQueryParameters;
+      Params: BaseChannelsParameters;
+    }>,
+  ): Promise<ResourceListResponse<ChannelObject>> {
+    const context = getExecutionContext(request);
+
+    const workspaces = (
+      await gr.services.workspaces.getAllForCompany(context.workspace.company_id)
+    ).map(a => a.id);
+
+    workspaces.unshift("direct");
+
+    const channels = await Promise.all(
+      workspaces.map(id =>
+        gr.services.channels.channels
+          .getAllChannelsInWorkspace(context.workspace.company_id, id)
+          .then(channels => {
+            if (id == "direct") {
+              return Promise.all(
+                channels.map(
+                  channel =>
+                    gr.services.channels.channels.includeUsersInDirectChannel(
+                      channel,
+                      context,
+                    ) as Promise<UsersIncludedChannel>,
+                ),
+              );
+            }
+            return channels;
+          }),
+      ),
+    );
+
+    let res: Channel[] = [];
+    channels.forEach(ch => {
+      res = res.concat(ch);
+    });
+
+    const activities = await Promise.all(
+      res.map(channel => gr.services.channels.channels.getChannelActivity(channel)),
+    );
+
+    const response: ChannelObject[] = [];
+
+    for (let i = 0; i < res.length; i++) {
+      const channel = res[i];
```

Comment:
Just for your info this is good stuff:
```
for (const channel of res){
 ...
}
```

Reply:
I know, but I need an index there, because I merge two arrays